yt/data_objects/profiles.py

```diff
@@ -288,7 +288,7 @@ def __getitem__(self, field):
                 fname = self.field_map.get(field[1], None)
                 if fname != field:
                     raise KeyError("Asked for field '{}' but only have data for "
-                                   "field '{}'".format(field, fname))
+                                   "fields '{}'".format(field, list(self.field_data.keys())))
```

Comment:
For reviewers, this was a separate bug I found. We were not printing the available fields but just the value of `fname`, which is usually None here.

Reply:
good catch!